config/initializers/devise.rb

```diff
@@ -0,0 +1,43 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
+  require 'devise/orm/active_record'
+
+  config.case_insensitive_keys = [:email]
+
+  config.strip_whitespace_keys = [:email]
+
+  config.skip_session_storage = [:http_auth]
+
+  config.stretches = Rails.env.test? ? 1 : 11
+
+  config.reconfirmable = true
+
+  config.expire_all_remember_me_on_sign_out = true
+
+  config.password_length = 6..128
+
+  config.email_regexp = /\A[^@\s]+@[^@\s]+\z/
+
+  config.reset_password_within = 6.hours
+
+  config.sign_out_via = :delete
+
+  config.omniauth :saml,
+                  :assertion_consumer_service_url     => "http://localhost:3003/users/auth/saml/callback",
+                  :assertion_consumer_service_binding => "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
+                  :issuer                             => Rails.configuration.x.saml_issuer,
+                  :idp_sso_target_url                 => Rails.configuration.x.idp_url,
+                  :idp_cert                           => File.read("#{Rails.root}/config/saml_cert.crt"),
+                  :name_identifier_format             => "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
+                  :authn_context                      => "http://idmanagement.gov/ns/assurance/loa/1",
+                  :allowed_clock_drift                => 60,
+                  :certificate                        => File.read('config/demo_sp.crt'),
+                  :private_key                        => File.read('config/demo_sp.key'),
+                  :security                           => {:authn_requests_signed => true,
```

Comment:
Since `demo_sp.key` is gitignored, an example key should be provided and copied over to `config/demo_sp.key` in `bin/setup`. See the -idp app for an example: https://github.com/18F/identity-idp/blob/master/bin/setup#L20-L22

Otherwise, `rake` or `rails` commands will fail with `No such file or directory @ rb_sysopen - config/demo_sp.key (Errno::ENOENT)`